src/Factory/ProductFactory.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+namespace Sylius\ElasticSearchPlugin\Factory;
+
+use ONGR\ElasticsearchBundle\Collection\Collection;
+use Sylius\Component\Core\Model\ChannelInterface;
+use Sylius\Component\Core\Model\ProductInterface;
+use Sylius\Component\Core\Model\ProductTranslationInterface;
+use Sylius\Component\Core\Model\ProductVariantInterface;
+use Sylius\Component\Locale\Model\LocaleInterface;
+use Sylius\Component\Resource\Model\TranslationInterface;
+use Sylius\ElasticSearchPlugin\Document\Attribute;
+use Sylius\ElasticSearchPlugin\Document\AttributeValue;
+use Sylius\ElasticSearchPlugin\Document\Price;
+use Sylius\ElasticSearchPlugin\Document\Product;
+use Sylius\ElasticSearchPlugin\Document\TaxonCode;
+
+final class ProductFactory implements ProductFactoryInterface
+{
+    /**
+     * @return Product
+     */
+    public function create()
+    {
+        return new Product();
+    }
+
+    /**
+     * @param ProductInterface $syliusProduct
+     * @param LocaleInterface $locale
+     * @param ChannelInterface $channel
+     *
+     * @return Product
+     */
+    public function createFromSyliusSimpleProductModel(ProductInterface $syliusProduct, LocaleInterface $locale, ChannelInterface $channel)
+    {
+        if (!$syliusProduct->isSimple()) {
```

Comment:
`Assert::true($syliusProduct->isSimple(), sprintf('Cannot create elastic search model from configurable product "%s" via this method.', $syliusProduct->getCode()));`